pyiron_base/generic/hdfio.py

```diff
@@ -291,40 +291,60 @@ def copy_to(self, destination, file_name=None, maintain_name=True):
             FileHDFio: FileHDFio object pointing to a file which now contains the same content as file of the current
                        FileHDFio object.
         """
+        def _internal_copy(source, source_path, target, target_path, maintain_flag):
+            """
+            Internal function to copy content of one HDF5 file to another or copy a group within the same HDF5 file. 
+            
+            Args:
+                source (h5py.File): HDF5 File object 
+                source_path (str): Path inside the source HDF5 file 
+                target (h5py.File): HDF5 File object
+                target_path (str): Path inside the target HDF5 file 
+                maintain_flag (bool): Maintain the same group name 
+            """
+            if maintain_flag:
+                try:
+                    target.create_group(target_path)
+                except ValueError:
+                    pass  # In case the copy_to() function failed previously and the group already exists. 
+            if target_path == "/":
+                source.copy(h5_path, "/") if source == target else source.copy(h5_path, target)
+                    source.copy(source_path, "/")
+                else:
+                    source.copy(source_path, target)
+            else:
+                if maintain_flag:
+                    source.copy(source_path, target[dest_path])
+                else:
+                    group_name_old = source_path.split("/")[-1]
+                    try:
+                        target.create_group("/tmp")
+                    except ValueError:
+                        pass
+                    source.copy(source_path, target["/tmp"])
+                    try:
+                        target.move("/tmp/" + group_name_old, target_path)
+                    except ValueError:
+                        del target[dest_path]
+                        target.move("/tmp/" + group_name_old, target_path)
+                    del target["/tmp"]
+
         if file_name is None:
             file_name = destination.file_name
         if self.file_exists:
-            with h5py.File(
-                self.file_name, mode="r", libver="latest", swmr=True
-            ) as f_source:
+            if destination.h5_path[0] == "/":
```

Comment:
Can pull the same one line trick here, but also need to get rid of the three lines below

```suggestion
            dest_path = destination.h5_path[1:] if destination.h5_path[0] == "/" else dest_path = destination.h5_path
```